### PR TITLE
feat: close unified_exec at end of turn

### PR DESCRIPTION
**Pull request by jif-oai**

### CODE REVIEW

File: codex-rs/core/src/unified_exec/session_manager.rs [hunk 1]
```diff
@@ -643,6 +643,11 @@ impl UnifiedExecSessionManager {
             entry.session.terminate();
         }
     }
+
+    pub(crate) async fn close_open_sessions(&self) {
+        // Terminating sessions ensures the exit watcher emits ExecCommandEnd events.
+        self.terminate_all_sessions().await;
```
Comment (pakrym-oai) [inline codex-rs/core/src/unified_exec/session_manager.rs]: Why have an alias for the same method?

File: codex-rs/core/src/tasks/mod.rs [hunk 2]
```diff
@@ -167,12 +168,18 @@ impl Session {
...
         let mut active = self.active_turn.lock().await;
-        if let Some(at) = active.as_mut()
+        let should_close_sessions = if let Some(at) = active.as_mut()
             && at.remove_task(&turn_context.sub_id)
         {
             *active = None;
-        }
+            true
+        } else {
+            false
```
Comment (pakrym-oai) [inline codex-rs/core/src/tasks/mod.rs:177]: do we need this condition? won't terminate_all_sessions just no-op when there are no sessions?

**Review by pakrym-oai — APPROVED** [review]

**Comment by pakrym-oai** [comment]
Can we add a test that starts long running process and asserts events (especially around TaskCompleted boundary)

**Comment by pakrym-oai** [comment]
And also asserts that the process is actually killed

File: codex-rs/core/src/tasks/mod.rs (hunk 2 above)
Comment (jif-oai) [reply codex-rs/core/src/tasks/mod.rs:177]: Yes because we can have tasks running
